Src/AutoFixture.NUnit3/InlineAutoDataAttribute.cs

```diff
@@ -0,0 +1,107 @@
+    [CLSCompliant(false)]
+    public class InlineAutoDataAttribute : Attribute, ITestBuilder
+    {
+        private readonly object[] _existingParameterValues;
+        private readonly IFixture _fixture;
+
+        /// <summary>
+        /// Construct a <see cref="InlineAutoDataAttribute"/>
+        /// with parameter values for test method
+        /// </summary>
+        public InlineAutoDataAttribute(params object[] arguments)
+            : this(new Fixture(), arguments)
+        {
+        }
+
+        /// <summary>
+        /// Construct a <see cref="InlineAutoDataAttribute"/> with an <see cref="IFixture"/> 
+        /// and parameter values for test method
+        /// </summary>
+        protected InlineAutoDataAttribute(IFixture fixture, params object[] arguments)
+        {
+            this._fixture = fixture;
+            this._existingParameterValues = arguments;
+        }
+
+        /// <summary>
+        ///     Construct one or more TestMethods from a given MethodInfo,
+        ///     using available parameter data.
+        /// </summary>
+        /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
+        /// <param name="suite">The suite to which the tests will be added.</param>
+        /// <returns>One or more TestMethods</returns>
+        public IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test suite)
+        {
+            var test = new NUnitTestCaseBuilder().BuildTestMethod(method, suite, this.GetParametersForMethod(method));
+
+            yield return test;
+        }
+
+        private TestCaseParameters GetParametersForMethod(IMethodInfo method)
+        {
+            try
+            {
+                var parameters = method.GetParameters();
+
+                var parameterValues = this.GetParameterValues(parameters);
+
+                return new TestCaseParameters(parameterValues.ToArray());
+            }
+            catch (Exception ex)
+            {
+                return new TestCaseParameters(ex);
+            }
+        }
+
+        /// <summary>
+        /// Get values for a collection of <see cref="IParameterInfo"/>
+        /// </summary>
+        private IEnumerable<object> GetParameterValues(IEnumerable<IParameterInfo> parameters)
+        {
+            return this._existingParameterValues.Concat(this.GetMissingValues(parameters));
+        }
+
+        private IEnumerable<object> GetMissingValues(IEnumerable<IParameterInfo> parameters)
+        {
+            var parametersWithoutValues = parameters.Skip(this._existingParameterValues.Count());
+
+            return parametersWithoutValues.Select(this.GetValueForParameter);
+        }
+
+        /// <summary>
+        /// Get value for an <see cref="IParameterInfo"/>
+        /// </summary>
+        private object GetValueForParameter(IParameterInfo parameterInfo)
+        {
+            CustomizeFixtureByParameter(this._fixture, parameterInfo);
```

Comment:
Why pass a class field as a parameter to another instance method on the same class?